llvm/include/llvm/ProfileData/SampleProfReader.h

```diff
@@ -413,6 +418,16 @@ class SampleProfileReader {
     if (It != Profiles.end())
       return &It->second;
 
+    if (FuncNameToProfNameMap && !FuncNameToProfNameMap->empty()) {
+      auto R = FuncNameToProfNameMap->find(FunctionId(Fname));
+      if (R != FuncNameToProfNameMap->end()) {
+        Fname = R->second.stringRef();
+        auto It = Profiles.find(FunctionId(Fname));
+        if (It != Profiles.end())
+          return &It->second;
+      }
+    }
```

Comment:
Can we avoid coupling between reader and the maps that are specific to profile matching? i.e. can we do this at caller side? 

Ideally, we avoid having `FuncNameToProfNameMap` in reader.